state_tracker.go

```diff
@@ -9,13 +9,118 @@ import (
 	"github.com/siddontang/go-mysql/mysql"
 )
 
+// StateTracker design
+// ===================
```

Comment:
I feel that it may be beneficial if we started leaving comments like this for all the major components. Thoughts?